interface/super/edit_layout.php

```diff
@@ -18,33 +18,29 @@
  * @link    http://www.open-emr.org
  */
 
+$fake_register_globals = false;
+$sanitize_all_escapes  = true;
+
 require_once("../globals.php");
 require_once("$srcdir/acl.inc");
 require_once("$srcdir/log.inc");
 
-$layouts = array(
-  'DEM' => xl('Demographics'),
-  'HIS' => xl('History'),
-  'FACUSR' => xl('Facility Specific User Information')
-);
-if ($GLOBALS['ippf_specific']) {
-    $layouts['GCA'] = xl('Abortion Issues');
-    $layouts['CON'] = xl('Contraception Issues');
-}
-
-// Include Layout Based Transaction Forms.
-$lres = sqlStatement("SELECT * FROM list_options " .
-  "WHERE list_id = 'transactions' AND activity = 1 ORDER BY seq, title");
-while ($lrow = sqlFetchArray($lres)) {
-    $layouts[$lrow['option_id']] = xl_list_label($lrow['title']);
-}
-
-// Include Layout Based Encounter Forms.
-$lres = sqlStatement("SELECT * FROM list_options " .
-  "WHERE list_id = 'lbfnames' AND activity = 1 ORDER BY seq, title");
-while ($lrow = sqlFetchArray($lres)) {
-    $layouts[$lrow['option_id']] = xl_list_label($lrow['title']);
+function collectLayoutNames($condition, $mapping='') {
+    global $layouts;
+    $gres = sqlStatement("SELECT grp_form_id, grp_title, grp_mapping " .
+      "FROM layout_group_properties WHERE " .
+      "grp_group_id = '' AND grp_activity = 1 AND $condition " .
```

Comment:
What are some examples of sue for $condition (to ensure this gets adequate sql escaping).

Reply:
See immediately below the function definition to see how it's called. $condition is just a literal in the call.